src/websocket-client.ts

```diff
@@ -57,6 +57,7 @@ export interface WSClientConfigurableOptions {
     agent?: any;
   };
   wsUrl?: string;
+  baseUrl?: string;
```

Comment:
Where did this come from? Don't think this applies for the WS client

Reply:
Please see #414. Without`baseUrl`, you cannot subscribe `SpotUserDataStream` on Testnet.

Reply:
This isn't the correct way of doing this unfortunately. Take a look at how testnet is initiated with the other embedded REST clients. I would strongly prefer if you undo this and we look at doing this properly in a separate PR (although I appreciate you looking to fix this too). Will check the rest in the meantime.

Reply:
As you asked, I pushed a new commit to undo this change.

>  Take a look at how testnet is initiated with the other embedded REST clients.

`getSpotRestClient()` function in line 756 of `websocket-client.js` file creates a new Rest Client. [Here on the pinned issue](https://github.com/tiagosiebler/binance/issues/209#issuecomment-1156071719), you described that to connect TestNet with Rest Client, we must provide `baseURL`.

Reply:
Ah I see. Admittedly that's more for the REST client, but since then other contributors have added a simpler way to configure testnet on some of the REST clients. We'll get the spot testnet working properly for the websocket client in a separate PR - most likely similar to how it's working for futures:
https://github.com/tiagosiebler/binance/blob/master/src/websocket-client.ts#L958-L963

But that's for a separate PR to get this one finished & merged. Will review the rest of this.

Reply:
Thanks, @tiagosiebler; I hope you'll provide a solution sooner because, currently, it is impossible to use WebSocket subscriptions on TestNet.